networkit/cpp/centrality/ApproxElectricalCloseness.cpp

```diff
@@ -0,0 +1,669 @@
+    G.forNodes([&](node u) { assert(status[u] == NodeStatus::NOT_VISITED); });
+#endif
+
+    // Topological order of biconnected components: tree of biconnected components starting from the
+    // root's biconnected component. If the root is in multiple biconnected components, we take one
+    // of them arbitrarily select one of them.
+    std::queue<std::pair<node, index>> q;
+    const auto &rootComps = bcc.getComponentsOfNode(root);
+    q.push({root, *(rootComps.begin())});
+
+    topOrder.reserve(bcc.numberOfComponents());
+    topOrder.insert(topOrder.begin(), rootComps.begin(), rootComps.end());
+
+    std::vector<count> distance(G.upperNodeIdBound());
+    status[root] = NodeStatus::VISITED;
+    rootEcc = 0;
+
+    do {
+        const auto front = q.front();
+        q.pop();
+        G.forNeighborsOf(front.first, [&](const node v) {
+            if (status[v] == NodeStatus::NOT_VISITED) {
+                distance[v] = distance[front.first] + 1;
+                rootEcc = std::max(rootEcc, distance[v]);
+                const auto &vComps = bcc.getComponentsOfNode(v);
+                for (const node vComponentIndex : vComps) {
+                    // Check if a new biconnected components has been found.
+                    if (vComponentIndex != front.second && biAnchor[vComponentIndex] == none
+                        && rootComps.find(vComponentIndex) == rootComps.end()) {
+                        // The anchor cannot be the root, because the anchor does not have a parent.
+                        // We handle biAnchor = none cases later.
+                        biAnchor[vComponentIndex] = (v == root) ? none : v;
+                        biParent[vComponentIndex] = front.second;
+                        topOrder.push_back(vComponentIndex);
+                    }
+                }
+                q.push({v, *(vComps.begin())});
+                status[v] = NodeStatus::VISITED;
+            }
+        });
+    } while (!q.empty());
+
+#ifndef NDEBUG
+    G.forNodes([&](node u) { assert(status[u] == NodeStatus::VISITED); });
+    assert(topOrder.size() == bcc.numberOfComponents());
+    assert(std::unordered_set<index>(topOrder.begin(), topOrder.end()).size()
+           == bcc.numberOfComponents());
+#endif
+}
+
+void ApproxElectricalCloseness::computeBFSTree() {
+    // Using thread 0's vector
+    auto &status = statusGlobal[0];
+    std::fill(status.begin(), status.end(), NodeStatus::NOT_VISITED);
+
+    std::queue<node> queue;
+    queue.push(root);
+    status[root] = NodeStatus::VISITED;
+
+    do {
+        const node currentNode = queue.front();
+        queue.pop();
+        G.forNeighborsOf(currentNode, [&](const node v) {
+            if (status[v] == NodeStatus::NOT_VISITED) {
+                status[v] = NodeStatus::VISITED;
+                queue.push(v);
+                bfsParent[v] = currentNode;
+            }
+        });
+    } while (!queue.empty());
+
+#ifndef NDEBUG
+    checkBFSTree();
+#endif
```

Comment:
Do not guard expensive checks behind NDEBUG. Add a separate definition to CMakeLists.txt to enable sanity checking.

Reply:
Something like `NETWORKIT_SANITY_CHECKS`.

Reply:
Sounds like a good idea, we might want set that flag to ON in some builds in our CI.

Reply:
Agreed.